### PR TITLE
[pcl] update to version 1.13.0

### PR DESCRIPTION
**Pull request by mvieth**
**Describe the pull request**

- #### What does your PR fix?
  Updates PCL to version 1.13.0, remove patches that are no longer needed

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  No changes, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

`Workaround-ICE-in-release.patch` removed because the fix is included in PCL 1.13.0 (https://github.com/PointCloudLibrary/pcl/commit/63cae5706eeec535f513e59286226ca8fa14340e).
`add_bigobj_option.patch` added because of the following change in PCL: https://github.com/PointCloudLibrary/pcl/commit/12b15359261f7248cfabc2c00d20d5f154c8f2c7 .
`fix-cmake_find_library_suffices.patch` removed, this is replaced by `-DPCL_ALLOW_BOTH_SHARED_AND_STATIC_DEPENDENCIES=ON` (see [also here](https://github.com/PointCloudLibrary/pcl/blob/master/cmake/pcl_options.cmake)).
`fix-error-C3052.patch` removed because the fix is included in PCL 1.13.0 (https://github.com/PointCloudLibrary/pcl/commit/cea5b2fd76bf8504701fecb272da4d18c8b9e02a).
`fix-find-libusb.patch` removed, should not be needed any more after changes in PCL: https://github.com/PointCloudLibrary/pcl/commit/c624e760c104d2af32cf8b03e95c34e4f20a3bcf and https://github.com/PointCloudLibrary/pcl/commit/374241c654c2daa188682ea21bad09b34738bedd .
`fix-find-qhull.patch` removed, should not be needed any more after changes in PCL: https://github.com/PointCloudLibrary/pcl/commit/c509b2ba0d7866b9d8353387cf9e8126eab03d93 .
`fix-namespace-cub.patch` removed because the fix is included in PCL 1.13.0 (https://github.com/PointCloudLibrary/pcl/commit/cea5b2fd76bf8504701fecb272da4d18c8b9e02a).
`fix-pkgconfig.patch` removed because the fix is included in PCL 1.13.0 (https://github.com/PointCloudLibrary/pcl/commit/b56e0f094bc5e2ca90320a5da21f49b647927672).
`no-absolute.patch` adapted, see also https://github.com/PointCloudLibrary/pcl/commit/bd975d59fe90580f09f928f67c20693b221aa021 .
`remove-broken-targets.patch` removed because I can't think of anything that might be a problem, and the PR that introduced the patch ( https://github.com/microsoft/vcpkg/pull/10042 ) doesn't give an explanation.

### CODE REVIEW

**Comment by mvieth** [comment]
@microsoft-github-policy-service agree

File: ports/pcl/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "pcl",
-  "version": "1.12.0",
-  "port-version": 7,
+  "version": "1.13.0",
+  "port-version": 0,
```
Comment (JonLiu1993) [inline ports/pcl/vcpkg.json:4]:
```suggestion
```

File: ports/pcl/portfile.cmake [hunk 2]
```diff
@@ -11,15 +11,11 @@ vcpkg_from_github(
         fix-numeric-literals-flag.patch
         pcl_config.patch
         pcl_utils.patch
-        remove-broken-targets.patch
+        # remove-broken-targets.patch
```
Comment (JonLiu1993) [inline ports/pcl/portfile.cmake:14]:
```suggestion
```

Comment (JonLiu1993) [reply ports/pcl/portfile.cmake:14]: @mvieth, thanks for the PR, please delete this patch here if it is no longer applicable.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
Please run command  "./vcpkg format-manifest ports/pcl/vcpkg.json" after finished the vcpkg.json file.

**Comment by mvieth** [comment]
@JonLiu1993 Thank you for your hints and suggestions. This PR is now ready for review. Please also see my explanations in the PR description, and please tell me if you want me to squash all commits. Unfortunately, I don't feel capable of migrating the existing deprecated function(s) in the portfile, so I left those unchanged.

**Review by JonLiu1993 — DISMISSED** [review]

**Comment by JonLiu1993** [comment]
@mvieth, when I tested the features by command "./vcpkg install pcl[*]:x64-windows" I get this error:
````
CMake Error at cmake/pcl_find_vtk.cmake:96 (message):
  Missing vtk modules: RenderingContextOpenGL2
Call Stack (most recent call first):
  CMakeLists.txt:393 (include)


-- Configuring incomplete, errors occurred!
See also "F:/Feature-test/pcl/vcpkg/buildtrees/pcl/x64-windows-rel/CMakeFiles/CMakeOutput.log".
See also "F:/Feature-test/pcl/vcpkg/buildtrees/pcl/x64-windows-rel/CMakeFiles/CMakeError.log".
````
Please take a look.

**Comment by mvieth** [comment]
@JonLiu1993 Please check again

**Review by BillyONeal — CHANGES_REQUESTED** [review]
```
.\vcpkg.exe install pcl[core,apps,cuda,examples,libusb,opengl,openni2,pcap,qt,simulation,tools,visualization,vtk]:x64-windows
```

fails with:

```
C:\vcpkg\buildtrees\pcl\src\89d532cf08-f1f6fba89c.clean\outofcore\tools\outofcore_viewer.cpp(118): fatal error C1083: Cannot open include file: 'vtkMutexLock.h': No such file or directory
```

[config-x64-windows-dbg-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/10503049/config-x64-windows-dbg-CMakeCache.txt.log)
[config-x64-windows-err.log](https://github.com/microsoft/vcpkg/files/10503050/config-x64-windows-err.log)
[config-x64-windows-out.log](https://github.com/microsoft/vcpkg/files/10503051/config-x64-windows-out.log)
[config-x64-windows-rel-CMakeCache.txt.log](https://github.com/microsoft/vcpkg/files/10503052/config-x64-windows-rel-CMakeCache.txt.log)
[error-logs-x64-windows.txt](https://github.com/microsoft/vcpkg/files/10503053/error-logs-x64-windows.txt)
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/10503054/install-x64-windows-dbg-out.log)
[stdout-x64-windows.log](https://github.com/microsoft/vcpkg/files/10503055/stdout-x64-windows.log)

Please investigate. Thanks for your contribution to vcpkg!

**Comment by mvieth** [comment]
@BillyONeal Thanks for your comment. I assume you tested with VTK's master branch? I added a patch to remove the include. The patch is already present on PCL's master branch (https://github.com/PointCloudLibrary/pcl/commit/b957fbee3bd837e8c569e0042691ff5c3baf2df7), but was not included in PCL 1.13.0.

**Comment by BillyONeal** [comment]
> I assume you tested with VTK's master branch?

I tested with the current copy of vtk vcpkg installs.

**Comment by BillyONeal** [comment]
No joy, this time:

```
C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1434~1.319\bin\Hostx64\x64\cl.exe   /TP -DH5_BUILT_AS_DYNAMIC_LIB -DKISSFFT_DLL_IMPORT=1 -DQT_CORE_LIB -DQT_DISABLE_DEPRECATED_BEFORE=0x050900 -DQT_GUI_LIB -DQT_OPENGLWIDGETS_LIB -DQT_OPENGL_LIB -DQT_WIDGETS_LIB -DUNICODE -DWIN32 -DWIN64 -D_ENABLE_EXTENDED_ALIGNED_STORAGE -D_UNICODE -D_WIN64 -Dkiss_fft_scalar=double -IC:\vcpkg\buildtrees\pcl\x64-windows-dbg\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\common\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\io\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\features\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\search\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\kdtree\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\octree\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\filters\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\keypoints\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\segmentation\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\sample_consensus\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\outofcore\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\stereo\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\geometry\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\surface\include -IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\visualization\include -external:IC:\vcpkg\installed\x64-windows\include\eigen3 -external:IC:\vcpkg\installed\x64-windows\include\openni2 -external:IC:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\recognition\include\pcl\recognition\3rdparty -external:IC:\vcpkg\installed\x64-windows\include -external:IC:\vcpkg\installed\x64-windows\include\vtk-9.2 -external:IC:\vcpkg\installed\x64-windows\include\Qt6\QtOpenGL -external:IC:\vcpkg\installed\x64-windows\include\Qt6 -external:IC:\vcpkg\installed\x64-windows\include\Qt6\QtCore -external:IC:\vcpkg\installed\x64-windows\share\Qt6\mkspecs\win32-msvc -external:IC:\vcpkg\installed\x64-windows\include\Qt6\QtGui -external:IC:\vcpkg\installed\x64-windows\include\Qt6\QtWidgets -external:IC:\vcpkg\installed\x64-windows\include\Qt6\QtOpenGLWidgets -external:W0 /nologo /DWIN32 /D_WINDOWS /W3 /utf-8 /GR /EHsc /MP  /bigobj /FS -DBOOST_ALL_DYN_LINK -DBOOST_ALL_NO_LIB /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1    -DBOOST_ALL_NO_LIB -D_SCL_SECURE_NO_WARNINGS -D_CRT_SECURE_NO_WARNINGS -DNOMINMAX -DPCL_ONLY_CORE_POINT_TYPES  -D__SSE4_2__ -D__SSE4_1__ -D__SSSE3__ -D__SSE3__ -D__SSE2__ -D__SSE__ -Zc:__cplusplus -permissive- -utf-8 -std:c++17 /showIncludes /Foexamples\surface\CMakeFiles\pcl_example_nurbs_fitting_closed_curve.dir\example_nurbs_fitting_closed_curve.cpp.obj /Fdexamples\surface\CMakeFiles\pcl_example_nurbs_fitting_closed_curve.dir\ /FS -c C:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\examples\surface\example_nurbs_fitting_closed_curve.cpp
C:\vcpkg\buildtrees\pcl\src\e50dc19856-e1542a945a.clean\surface\include\pcl\surface\3rdparty\opennurbs\opennurbs_system.h(147): fatal error C1189: #error:  do not define WIN32 for x64 builds
```

Can you please try running

x-set-installed pcl[core,apps,cuda,examples,libusb,opengl,openni2,pcap,qt,simulation,tools,visualization,vtk]:x64-windows

and let me know when you've got that working?

**Comment by mvieth** [comment]
@BillyONeal Please review again.
Unfortunately I don't have a Windows system to test on, but @larshg was so kind to do the testing. And now all problems should be fixed :smile:

**Comment by JonLiu1993** [comment]
`.\vcpkg.exe install pcl[core,apps,cuda,examples,libusb,opengl,openni2,pcap,qt,simulation,tools,visualization,vtk]:x64-windows `tested successfully.

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Comment by mvieth** [comment]
Thanks for the reviews!

**Pull request merged**